added test for pair_programming in Git backend

imnitishng opened the pull request:
Added tests for `pair_programming` in Git backend to increase coverage
Signed-off-by: Nitish Gupta <imnitish.ng@gmail.com>

valeriocos commented on tests/base.py:
```diff
@@ -274,10 +274,18 @@ def _test_study(self, test_study):
-                                                                 db_user=self.db_user,
-                                                                 db_password=self.db_password,
-                                                                 db_projects_map=self.db_projects_map)
+        # different arguments to test pair_programming
```
Why you want to test the pair programming with the demography study? 

Pair programming could be calculated in a different test, leveraging on the _test_raw_to_enrich method.

imnitishng replied:
Okay. I get the point. I am sorry I did not realize this, I will soon rework this and send the PR.

valeriocos replied:
No worries, thank you @imnitishng :)

valeriocos commented on tests/data/git.json:
```diff
@@ -61,16 +61,20 @@
     "tag": "/tmp/perceval_mc84igfc/gittest",
     "timestamp": 1518165202.694393,
     "updated_on": 1344965413.0,
+    "git_uuid": "43f217b2f678a5691fdbc5c6c5302243e79e5a90",
```
why is it needed to include the attribute `git_uuid`? This comment applies also to the other changes on this attribute

imnitishng replied:
The `git_uuid` attribute is created [here](https://github.com/chaoss/grimoirelab-elk/blob/39596eae4441815fda65ac8374e7fb4cd8781088/grimoire_elk/enriched/git.py#L110)) once `pair_programming` parameter is set to True in the GitEnrich class. 
This function is called several times while uploading data to elasticsearch, [for instance here](https://github.com/chaoss/grimoirelab-elk/blob/39596eae4441815fda65ac8374e7fb4cd8781088/grimoire_elk/enriched/enrich.py#L1553) so the attribute was needed.

valeriocos replied:
I see, thank you for the pointers. 

The test data is basically the data that comes from Perceval, and it is modified to reflect data that comes directy from Perceval (e.g., a missing attribute, a multi authored commit). The `git_uuid` is something that is added during the enrichment phase, thus it shouldn't appear in the test data.

Let me know if you have questions, thanks!

imnitishng replied:
Okay I get it. Thank You! But while working on this PR I saw absence of this attribute caused some tests to fail. Nevermind I will work on the new approach and get to you if I encounter any problems or doubts related to this attribute.

valeriocos replied:
Ok, thanks!

valeriocos commented:
please @imnitishng ping me when the PR is ready for review (or let me know if you need some help), thanks!

imnitishng commented:
Okay, @valeriocos, sorry for the delay I have been busy with gitter backend, I will send this PR tomorrow.

valeriocos commented:
no worries @imnitishng , thank you for the quick reply!

imnitishng commented:
Hi @valeriocos, please have a look I have fixed the `pair_programming` issue, however 2 new demographic tests for `gitter` and `git` have been failing, I tried but couldn't find the issue, because the files I have changed don't seem to affect them. I think the issue is with my machine, so I'll wait and check if the tests here pass successfully.

![Screenshot from 2020-04-09 23-23-55](https://user-images.githubusercontent.com/35267629/78925583-501f7a00-7ab9-11ea-8601-dd524b838516.png)

imnitishng commented:
Okay, checks seem to pass here, IDK why they failed on my machine, anyway please review the PR when you get the time.

imnitishng commented:
Okay, checks seem to pass here, IDK why they failed on my machine, anyway please review the PR when you get the time.

valeriocos commented:
The tests fail because thre is already an index in your machine using the demographics alias. If you remove that alias (or delete the index) , the tests will pass.

imnitishng commented:
@valeriocos, please have look.
